tests/src/test/java/org/eclipse/hono/tests/mqtt/MqttPublishTestBase.java

```diff
@@ -869,13 +870,14 @@ protected void doTestUploadMessages(
         customizeConnectedClient();
 
         final long start = System.currentTimeMillis();
+        final long timeout = start + getTimeToWait();
         while (messageCount.get() < MESSAGES_TO_SEND) {
             final CountDownLatch messageHandlingCompleted = new CountDownLatch(errorMessagesExpected ? 2 : 1);
             context.runOnContext(go -> {
                 final Buffer msg = Buffer.buffer("hello " + messageCount.getAndIncrement());
                 sender.apply(msg).onComplete(sendAttempt -> {
                     if (sendAttempt.failed()) {
-                        LOGGER.error("error sending message {}", messageCount.get(), sendAttempt.cause());
+                        LOGGER.trace("error sending message {}", messageCount.get(), sendAttempt.cause());
```

Comment:
publishing the message should always work, shouldn't it? Thus, if it doesn't, I'd consider this an event that is worth logging at *error* level ...

Reply:
I thought there a test cases that would fail the send attempt itself. I will return back to error.
But if it is unexpected then shouldn't test fail on that condition?